examples/loki/nurf_data.py

```diff
@@ -102,120 +102,173 @@ def nurf_file_creator(loki_file, path_to_loki_file, data):
-
-        uv_bkg = grp_uv.create_dataset('uv_background',
-                                       data=data['UV_background'],
-                                       shape=data['UV_background'].shape,
-                                       dtype=np.float32)
-        uv_bkg.attrs['long name'] = 'uv_background'
-
-        uv_int0 = grp_uv.create_dataset('uv_intensity0',
-                                        data=data['UV_intensity0'],
-                                        shape=data['UV_intensity0'].shape,
-                                        dtype=np.float32)
-        uv_int0.attrs['long name'] = 'uv_intensity0'
-
-        uv_spectra = grp_uv.create_dataset('uv_spectra',
-                                           data=data['UV_spectra'],
-                                           shape=data['UV_spectra'].shape,
-                                           dtype=np.float32)
-        uv_spectra.attrs['long name'] = 'uv_spectra'
-
-        uv_wavelength = grp_uv.create_dataset('uv_wavelength',
-                                              data=data['UV_wavelength'],
+        grp_uv.attrs["NX_class"] = 'NXdata'
+        
+        # subgroup for uv all data (sample, dark, reference)
+        uv_signal=grp_uv.create_group("uv_all_data")
+        uv_signal_data=uv_signal.create_dataset('data', data=uv_all_data,
+                                           shape=uv_all_data.shape)
+        uv_signal_data.attrs['long_name']= 'uv_all_data'
+        uv_signal_data.attrs['units']= ''
+        uv_signal_data.attrs['signal']= 'data'  #indicate that the main signal is data 
+        uv_signal_data.attrs['axes']= [ "wavelength", "." ]   #see example in NXdata, time as x-axis is first entry
+        uv_signal_image_key=uv_signal.create_dataset('image_key',data=uv_image_key,shape=uv_image_key.shape, dtype=np.int32)
+       
+     
+        # uv_wavelength
+        uv_wavelength_data=grp_uv.create_dataset('uv_wavelength', data=data['UV_wavelength'],
                                               shape=data['UV_wavelength'].shape,
                                               dtype=np.float32)
-        uv_wavelength.attrs['units'] = 'nm'  # TODO: unit to be verified
-        uv_wavelength.attrs['long name'] = 'uv_wavelength'
+        
+        uv_wavelength_data.attrs['units'] = 'nm'  # TODO: unit to be verified
+        uv_wavelength_data.attrs['long name'] = 'uv_wavelength'
+                
+                
+        # uv_integration_time
+        uv_inttime_data=grp_uv.create_dataset("uv_integration_time",data=data['UV_IntegrationTime'],
+                                           shape=data[
+                                               'UV_IntegrationTime'].shape, 
+                                           dtype=np.int32)
+                   
+        uv_inttime_data.attrs['long_name'] = 'uv_integration_time'
+        uv_inttime_data.attrs['units'] = 'us'  # TODO: unit to be verified, currently in micro-seconds
```

Comment:
What is this? Not an axis for the data? An auxiliary signal? As it is now, I don't see how this could be loaded.

Reply:
Integration time could be considered as an auxiliary signal I guess. We need to record the exposure time of each spectrum. I am unsure how to add `auxiliary_signals` to the code.  NXdata does not show an example.

Reply:
```python
grp_uv.attrs['auxiliary_signals'] = ['uv_integration_time']
```

Reply:
> ```python
> grp_uv.attrs['auxiliary_signals'] = ['uv_integration_time']
> ```

Okay! But the value for `uv_integration_time` remains a dataset, please?

Reply:
Yes, this simply says "this dataset contains values for an auxiliary signal".